pkg/workloadmeta/collectors/internal/process/process.go

```diff
@@ -0,0 +1,145 @@
+	ddConfig := config.Datadog
+
+	wlmExtractor := workloadmetaExtractor.NewWorkloadMetaExtractor(ddConfig)
+
+	processData := checks.NewProcessData(ddConfig)
+	processData.Register(wlmExtractor)
+
+	c = &collector{
+		ddConfig:        ddConfig,
+		wlmExtractor:    wlmExtractor,
+		grpcServer:      workloadmetaExtractor.NewGRPCServer(ddConfig, wlmExtractor),
+		processData:     processData,
+		collectionClock: clock.New(),
+		pidToCid:        make(map[int]string),
+	}
+	return c
+}
+
+// Compile time check to ensure that `collector` implements `workloadmeta.Collector`.
+var _ workloadmeta.Collector = (*collector)(nil)
+
+type collector struct {
+	ddConfig config.ConfigReader
+
+	processData *checks.ProcessData
+
+	wlmExtractor *workloadmetaExtractor.WorkloadMetaExtractor
+	grpcServer   *workloadmetaExtractor.GRPCServer
+
+	pidToCid map[int]string
+
+	collectionClock clock.Clock
+}
+
+func (c *collector) Start(ctx context.Context, store workloadmeta.Store) error {
+	if enabled, err := Enabled(c.ddConfig); !enabled {
+		return err
+	}
+
+	err := c.grpcServer.Start()
+	if err != nil {
+		return err
+	}
+
+	collectionTicker := c.collectionClock.Ticker(
+		c.ddConfig.GetDuration("workloadmeta.local_process_collector.collection_interval"),
+	)
+
+	filter := workloadmeta.NewFilter([]workloadmeta.Kind{workloadmeta.KindContainer}, workloadmeta.SourceAll, workloadmeta.EventTypeAll)
+	containerEvt := store.Subscribe(collectorId, workloadmeta.NormalPriority, filter)
+
+	go c.run(ctx, store, containerEvt, collectionTicker)
+
+	return nil
+}
+
+func (c *collector) run(ctx context.Context, store workloadmeta.Store, containerEvt chan workloadmeta.EventBundle, collectionTicker *clock.Ticker) {
+	defer c.grpcServer.Stop()
+	defer store.Unsubscribe(containerEvt)
+	defer collectionTicker.Stop()
+
+	for {
+		select {
+		case evt := <-containerEvt:
+			c.handleContainerEvent(evt)
+		case <-collectionTicker.C:
+			err := c.processData.Fetch()
+			if err != nil {
+				_ = log.Error("Error fetching process data:", err)
+			}
+		case <-ctx.Done():
+			log.Infof("The %s collector has stopped", collectorId)
+			return
+		}
+	}
+}
+
+// Pull is unused at the moment used due to the short frequency in which it is called.
+// In the future, we should use it to locally in workload-meta.
+func (c *collector) Pull(_ context.Context) error {
+	return nil
+}
+
+func (c *collector) handleContainerEvent(evt workloadmeta.EventBundle) {
+	defer close(evt.Ch)
+
+	for _, evt := range evt.Events {
+		ent := evt.Entity.(*workloadmeta.Container)
+		switch evt.Type {
+		case workloadmeta.EventTypeSet:
+			// Should be safe, even on windows because PID 0 is the idle process and therefore must always belong to the host
+			if ent.PID != 0 {
+				c.pidToCid[ent.PID] = ent.ID
+			}
+		case workloadmeta.EventTypeUnset:
+			delete(c.pidToCid, ent.PID)
+		}
+	}
+
+	c.wlmExtractor.SetLastPidToCid(c.pidToCid)
+}
+
+func Enabled(cfg config.ConfigReader) (bool, error) {
```

Comment:
[minor] Add a comment to provide some context around these config checks.

Reply:
Done 👍🏻